lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -199,6 +199,162 @@ static void printNoIOOp(Operation *op, OpAsmPrinter *printer) {
   printer->printOptionalAttrDict(op->getAttrs());
 }
 
+//===----------------------------------------------------------------------===//
+// spv.AccessChainOp
+//===----------------------------------------------------------------------===//
+
+// Function to extract an actual value from ConstantOp.
+static std::pair<StringRef, bool> extractValueFromConstOp(Operation *op,
+                                                          int32_t &indexValue) {
+  // spv.constant operation contains kValueAttrName.
+  auto valueAttr = op->getAttr(kValueAttrName);
+  if (!valueAttr) {
+    return {
+        "index must be an integer constant to access an element of StructType",
+        false};
+  }
+
+  auto integerValueAttr = valueAttr.dyn_cast<IntegerAttr>();
+  if (!integerValueAttr) {
+    return {"expected an integer constant for index", false};
+  }
+
+  indexValue = integerValueAttr.getInt();
+  return {"", true};
+}
+
+static ParseResult parseAccessChainOp(OpAsmParser *parser,
+                                      OperationState *state) {
+  OpAsmParser::OperandType ptrInfo;
+  SmallVector<OpAsmParser::OperandType, 4> indicesInfo;
+  spirv::StorageClass resultStorageClass;
+  Type type;
+  llvm::SMLoc indicesLocation;
+  int32_t index;
+  Type indicesType = parser->getBuilder().getIndexType();
+
+  if (parser->parseOperand(ptrInfo) ||
+      parser->getCurrentLocation(&indicesLocation) ||
+      parser->parseOperandList(indicesInfo, OpAsmParser::Delimiter::Square) ||
+      parser->parseColonType(type) ||
+      parser->resolveOperand(ptrInfo, type, state->operands) ||
+      parser->resolveOperands(indicesInfo, indicesType, state->operands)) {
+    return failure();
+  }
+
+  if (!indicesInfo.size()) {
```

Comment:
Can we extract this part (L245 - L287) and share them between the parser and the verifier? There is `mlir::emitError` that we can use; we just need to pass in the `Location` for error reporting: `return emitError(loc, "'spv.AccessChain' op ...")`.

Reply:
Updated, but have some concerns about an error message, in this case error message points to ssa value and not to the proccessing operation. I'm not sure how to convert llvmLoc to Location.
Thanks.